src/main/scala/viper/gobra/translator/encodings/channels/ChannelEncoding.scala

```diff
@@ -0,0 +1,248 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.encodings.channels
+
+import org.bitbucket.inkytonik.kiama.==>
+import viper.gobra.ast.{internal => in}
+import viper.gobra.reporting.Source
+import viper.gobra.theory.Addressability
+import viper.gobra.theory.Addressability.{Exclusive, Shared}
+import viper.gobra.translator.Names
+import viper.gobra.translator.encodings.LeafTypeEncoding
+import viper.gobra.translator.interfaces.Context
+import viper.gobra.translator.util.ViperWriter.CodeWriter
+import viper.silver.{ast => vpr}
+
+class ChannelEncoding extends LeafTypeEncoding {
+
+  import viper.gobra.translator.util.TypePatterns._
+  import viper.gobra.translator.util.ViperWriter.CodeLevel._
+
+  /**
+    * Translates a type into a Viper type.
+    */
+  override def typ(ctx : Context) : in.Type ==> vpr.Type = {
+    case ctx.Channel(t) / m =>  m match {
+      case Exclusive => vpr.Int // for now we simply map it to Viper Ints
+      case Shared => vpr.Ref
+    }
+  }
+
+  /**
+    * Encodes expressions as values that do not occupy some identifiable location in memory.
+    *
+    * R[ dflt(channel[T]) ] -> 0
+    * R[ nil : channel[T] ] -> 0
+    * R[ <- (c: channel[T] ] ->
+    *   exhale acc([c].RecvChannel(), wildcard)
+    *   exhale [c].RecvGivenPerm()()
+    *   var res [ T ]
+    *   inhale [c].RecvChannel()
```

Comment:
If I see this right, then the permission amount increases everytime something is received.
How about `exhale acc([c].RecvChannel(), perm([c].RecvChannel()))` instead?

Reply:
According to Martin, this permission is duplicable and this treatment should simplify the encoding (I don‘t know which version would lead to more performance).

Reply:
Yeah, yeah, we can think about that at a later point in time. I also do not know what is better for performance. We should evaluate this at some point and then make changes accordingly.